comp/trace/agent/impl/agent.go

```diff
@@ -86,6 +86,14 @@ func (c component) SendStatsPayload(p *pb.StatsPayload) {
 	c.Agent.StatsWriter.SendPayload(p)
 }
 
+func (c component) GetHTTPHandler(endpoint string) http.Handler {
+	c.Agent.Receiver.BuildHandlers()
+	if v, ok := c.Agent.Receiver.Handlers[endpoint]; ok {
+		return v
+	}
+	return nil
```

Comment:
Does the caller is supposed to handle `nil`?

Reply:
Yes, the caller handles the `nil`